packages/build-utils/src/convert-runtime-to-plugin.ts

```diff
@@ -179,6 +180,81 @@ export function convertRuntimeToPlugin(
       await fs.ensureDir(dirname(entry));
       await fs.copy(handlerFile.fsPath, entry);
 
+      // For compiled languages, the launcher file will be binary and therefore
+      // won't try to import a user-provided request handler (instead, it will
+      // contain it). But for interpreted languages, the launcher might try to
+      // load a user-provided request handler from the source file instead of bundling
+      // it, so we have to adjust the import statement inside the launcher to point
+      // to the respective source file. Previously, Legacy Runtimes simply expected
+      // the user-provided request-handler to be copied right next to the launcher,
+      // but with the new File System API, files won't be moved around unnecessarily.
+      if (handlerHasImport) {
+        const { fsPath } = handlerFile;
+        const encoding = 'utf-8';
+
+        // This is the true directory of the user-provided request handler in the
+        // source files, so that's what we will use as a import path in the launcher.
+        const locationPrefix = relative(entry, outputPath);
+
+        let handlerContent = await fs.readFile(fsPath, encoding);
+
+        const importPaths = [
+          // This is the full entrypoint path, like `./api/test.py`
+          `./${entrypoint}`,
+          // This is the entrypoint path without extension, like `api/test`
+          entrypoint.replace(ext, ''),
+        ];
+
+        // Generate a list of regular expressions that we can use for
+        // finding matches, but only allow matches if the import path is
+        // wrapped inside single (') or double quotes (").
+        const patterns = importPaths.map(path => {
+          // eslint-disable-next-line no-useless-escape
+          return new RegExp(`('|")${path.replace(/\./g, '\\.')}('|")`, 'g');
```

Comment:
Let's add some tests so we can refactor this implementation in the futuer

Reply:
As mentioned before, the output currently isn't yet stable and working, so I will defer adding tests until it's working, because the API is still changing quite dramatically, as we've seen yesterday yet again.